options.go

```diff
@@ -45,9 +53,54 @@ func WithCredentialsJSON(p []byte) DialerOption {
 	}
 }
 
+// WithDefaultDialOption returns a DialerOption that specifies the default DialOptions used.
+func WithDefaultDialOptions(opts ...DialOption) DialerOption {
+	return func(d *dialerConfig) {
+		d.dialOpts = append(d.dialOpts, opts...)
+	}
+}
+
 // WithTokenSource returns a DialerOption that specifies an OAuth2 token source to be used as the basis for authentication.
 func WithTokenSource(s oauth2.TokenSource) DialerOption {
 	return func(d *dialerConfig) {
 		d.sqladminOpts = append(d.sqladminOpts, apiopt.WithTokenSource(s))
 	}
 }
+
+// A DialOption is an option for configuring how a Dialer's Dial call is executed.
+type DialOption func(d *dialCfg)
+
+type dialCfg struct {
+	tcpKeepAlive time.Duration
+	ipType       string
+}
+
+// DialOptions turns a list of DialOption instances into an DialOption.
+func DialOptions(opts ...DialOption) DialOption {
```

Comment:
Is this function a convenience for the client or for the library maintainers?

Reply:
Convenience for the user. If you want to have a bunch of "Default" options, you would normally make a slice. But with variadic args that means you have do do something like this: 

```go
defaultOptions := []Option{Timeout(5 * time.Second)}

server1 := NewServer(":8080", append(defaultOptions, MaxConnections(10))...)
server2 := NewServer(":8080", append(defaultOptions, RateLimit(10, time.Minute))...)
```

With this convenience function, you can do something like this, which is (imo) more transparent:
```golang
defaultOptions := Options(Timeout(5 * time.Second))

server1 := NewServer(":8080", defaultOptions, MaxConnections(10))
server2 := NewServer(":8080", defaultOptions, RateLimit(10, time.Minute))
```

Reply:
That looks very nice. Thanks for the clarification.